sale_stock_picking_note/models/sale_order.py

```diff
@@ -22,9 +22,15 @@ class SaleOrder(models.Model):
         readonly=False,
     )
 
-    @api.depends("partner_id")
+    @api.depends("partner_shipping_id")
```

Comment:
According to the code, these should be the dependencies, right?

```suggestion
    @api.depends("partner_id", "partner_shipping_id")
```

Reply:
I think it's not neccesary, if you change partner, partner_shipping also changes by this compute https://github.com/odoo/odoo/blob/16.0/addons/sale/models/sale_order.py#L345

Reply:
You shouldn't rely on "external logic" for your depends. If the defining field is `partner_shipping_id`, then put this field as it depends. If tomorrow Odoo (or your customizations) decides that changing partner doesn't change the delivery address, your code will continue working without problems.

Reply:
Ok, I changed it

Reply:
Apart from that, if you define dependencies appropriately, you indirectly instruct Odoo the order in which it should process the computes involved in a change.